### ISSUE TITLE
Error: Failed to fetch API

### ISSUE THREAD

**Issue by salcode** [WebDevStudios/nextjs-wordpress-starter#17]
When I run `yarn dev`, I'm getting the following failure.

I wonder if this is a result of updating WP GraphQL on the WordPress Backend (#16)

```
[
  {
    message: 'Cannot query field "sourceUrl" on type "NodeWithFeaturedImageToMediaItemConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "name" on type "NodeWithAuthorToUserConnectionEdge". Did you mean "node"?',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "firstName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "lastName" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  },
  {
    message: 'Cannot query field "avatar" on type "NodeWithAuthorToUserConnectionEdge".',
    extensions: { category: 'graphql' },
    locations: [ [Object] ]
  }
]
Error: Failed to fetch API
    at fetchAPI (webpack-internal:///./lib/api.js:36:11)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async getAllPostsForHome (webpack-internal:///./lib/api.js:73:16)
    at async getStaticProps (webpack-internal:///./pages/index.js:110:20)
    at async renderToHTML (/Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/next-server/server/render.js:26:109)
    at async /Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/next-server/server/next-server.js:76:329
    at async __wrapper (/Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/lib/coalesced-function.js:1:330)
    at async DevServer.renderToHTMLWithComponents (/Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/next-server/server/next-server.js:99:365)
    at async DevServer.renderToHTML (/Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/next-server/server/next-server.js:100:255)
    at async DevServer.renderToHTML (/Users/sal/wds/nextjs-wordpress-starter/node_modules/next/dist/server/next-dev-server.js:34:578)
```

**Comment by salcode** [comment]
Thinking about this, if we have the following options now that the `author` object returned by WP GraphQL has changed

## Option 1: Pass contents of node

```
<Avatar author={author.node} />

function Avatar({author}) {
  return author.avatar.url;
}
```

## Option 2: Pass object that contains node

```
<Avatar author={author} />

function Avatar({author}) {
  return author.node.avatar.url;
}
```

I think `Option 2: Pass object that contains node` is the better choice (though `Option 1: Pass contents of node` is the approach I was trying to take yesterday but in retrospect I think it was the wrong move)

**Comment by gregrickaby** [comment]
@salcode I'm good w/ Option 2

**Comment by salcode** [comment]
@dcooney I know you have essentially this same fix in commit abc5ca9cb1b54f64753c359dda37e50e30d2072a in the [feature/menus](https://github.com/WebDevStudios/nextjs-wordpress-starter/tree/feature/menus) branch. Unless we're ready to merge that branch into `main` (which I don't think we are yet). I'd like to fix this here in `main` and then we can merge these changes in the `feature/menus` branch.
